src/Core/Framework/Util/Random.php

```diff
@@ -65,30 +77,41 @@ public static function getString(int $length, ?string $charlist = null): string
             $result .= $charlist[$pos];
         }
 
+        /** @var non-empty-string $result phpstan does not understand that it has been filled by the loop */
```

Comment:
this is a bug to me :thinking: see https://phpstan.org/r/393e0433-f5a3-4ac4-832a-23068671f17d

I will report it

Reply:
maybe we want to deactivate that config again :thinking: are at least changing the comment and put an assert instead